### DISCUSSION TITLE
routing performance

### DISCUSSION THREAD

**Discussion by ITwrx** [luckyframework/lucky#1360] [General] [answered]
I know it's just one benchmark 
https://github.com/the-benchmarker/web-frameworks

but is there some reason why it's flawed, or not representative of the routing speed of lucky? I'm looking for a viable replacement for laravel, and speed is one of the main reasons. The linked benchmark shows lucky as only 3 times faster than laravel while amber and kemal are 625x faster than laravel and 204x faster than lucky. If this is representative, why, and will it be addressed anytime soon? or can't be addressed due to design choices that are not expected to change?
thanks

**Comment by matthewmcgarvey** [comment, marked as answer]
@ITwrx we recommend referring to TechEmpower's benchmarks instead. Here's the latest run of them and shows Lucky having relatively close performance with the other crystal frameworks https://www.techempower.com/benchmarks/#section=test&runid=627e33f6-2a02-40f5-b377-a36e65f69e92

**Reply by jwoertink** [reply]
Oh yeah! I always forget we're on that 😄  Good call

**Comment by jwoertink** [comment]
Ah darn. I'm actually a maintainer on that benchmark repo. There's an issue with how Lucky is setup on there, and I didn't want to fix it because at the time, we were dead last and I figured "hey, if we can't be first, then we're gonna be last!" 😂  But I guess now that we're not last, it actually does look like Lucky is slow (which it's not).

One issue with that benchmark repo is that it tests how fast a framework can render a `HEAD` call (or in the case of [Laravel](https://github.com/the-benchmarker/web-frameworks/blob/master/php/laravel/app/Http/Controllers/ApplicationController.php#L11) and empty string) which isn't very helpful. Once you add in a little logic, a middleware stack, and a DB call, all of the numbers really start to get a bit crazy.

Lucky is currently on par with the speed that Amber shows. I'll take a look at the benchmark, but in the meantime I'd highly recommend taking a few minutes to just spin up a Lucky app, then just see if you like the flow. We can help you out with benchmarking it if there's something in particular you need to ensure is "blazing fast". 

Welcome to Lucky! 🎉

**Comment by ITwrx** [comment]
whoops, i forgot to look back at the techempower benchmarks... thanks both for all the info. that's great to hear. i actually already started with Nim, and while i like the syntax and language a lot, i'm starting to think the ecosystem (web in particular) is not far enough along for my needs at the moment. I'm not low level enough yet to fix or work around some of the issues in a reasonable amount of time, so now i'm reconsidering Crystal and looking at it's frameworks. Coming from laravel (meaning i'm spoiled in some ways), lucky's seemingly thorough docs are a big plus.
